src/main/webapp/components/workspaces/workspaces.js

```diff
@@ -59,6 +68,181 @@ const workspaceById = gql`
     }
   }
 `
+const queryAttributes = gql`
+  fragment QueryAttributes on MetacardAttributes {
+    title
+    cql
+    metacard_tags
+    metacard_type
+  }
+`
+
+const useCreateQuery = workspaceId => {
+  const [queryId, setQueryId] = React.useState('')
+  const mutation = gql`
+    mutation CreateQuery($attrs: MetacardAttributesInput!) {
+      createMetacard(attrs: $attrs) {
+        id: id
+        ...QueryAttributes
+      }
+    }
+    ${queryAttributes}
+  `
+
+  const saveMutation = gql`
+    mutation SaveWorkspace($id: ID!, $attributes: Json!) {
+      saveMetacardFromJson(id: $id, attributes: $attributes) {
+        ...WorkspaceAttributes
+      }
+    }
+    ${workspaceAttributes}
+  `
+
+  const [save] = useMutation(saveMutation)
+
+  const [create] = useMutation(mutation, {
+    onCompleted: data => {
+      save({
+        variables: {
+          id: workspaceId,
+          attributes: {
+            queries: [data.createMetacard.id],
+            metacard_type: 'workspace',
+          },
+        },
+        refetchQueries: ['WorkspaceById'],
+      }),
+        setQueryId(data.createMetacard.id)
+    },
+  })
+
+  return [create, queryId]
+}
+
+const TextSearchForm = props => {
+  const [textValue, setTextValue] = React.useState('')
+  const [title, setTitle] = React.useState('')
+  const [create, queryId] = useCreateQuery(props.workspaceId)
+  const { runQuery } = React.useContext(WorkspaceContext)
+
+  const onCreate = async () => {
+    const filter = {
+      property: 'anyText',
+      type: 'ILIKE',
+      value: textValue,
+    }
+
+    const cql = transformFilterToCQL(filter)
+    try {
+      await create({
+        variables: {
+          attrs: {
+            title: title,
+            cql: cql,
+            metacard_tags: ['query'],
+            metacard_type: 'metacard.query',
+          },
+        },
+      })
+    } catch (err) {
+      //eslint-disable-next-line
+      console.err('Error creating search or updating workspace: ', err)
+    }
+  }
+
+  // When the query id actually exists, run it.
+  React.useEffect(
+    () => {
+      if (queryId) {
+        // Get query
+        runQuery(queryId)
+      }
+    },
+    [queryId, runQuery]
+  )
+
+  const onClickSearch = async () => {
+    await onCreate(true)
```

Comment:
Don't have to wait anymore because nothing is happening afterwards.

```suggestion
  const onClickSearch = () => {
    onCreate(true)
```

Reply:
Also looks like you don't need the `true` argument because onCreate doesn't take any arguments